src/kotlin/flwr/src/main/java/dev/flower/android/Worker.kt

```diff
@@ -0,0 +1,51 @@
+package dev.flower.android
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.CoroutineWorker
+import androidx.work.Data
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.PeriodicWorkRequest
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+class FlwrWorker(private val context: Context,
+                 workerParams: WorkerParameters,
+                 private val flwrClient: Client,
+                 private val grpcRere: Boolean) : CoroutineWorker(context, workerParams) {
+    override suspend fun doWork(): Result {
+        if (grpcRere) {
+            inputData.getString("serverAddress")?.let { startFlowerRere(it, false, flwrClient) }
+        } else {
+            inputData.getString("serverAddress")?.let { startClient(it, false, flwrClient) }
+        }
+        val workManager: WorkManager = WorkManager.getInstance(context)
+        workManager.cancelWorkById(id)
+        return Result.success()
+    }
+
+    fun startFlwrPeriodicWorker(interval: Long, serverAddress: String) {
```

Comment:
@danielnugraha can you add documentation which clarifies the format of `serverAddress` and TimeUnit for `interval` this function expects?

Reply:
done.